src/crypto/ocb_internal.cc

```diff
@@ -358,30 +362,82 @@
 #if USE_OPENSSL_AES
 /*---------------*/
 
-#include <openssl/aes.h>                            /* http://openssl.org/ */
+#include <openssl/evp.h>                            /* http://openssl.org/ */
 
 namespace ocb_aes {
 
-typedef AES_KEY KEY;
+typedef EVP_CIPHER_CTX KEY;
 
-static KEY *KEY_new() { return new KEY; }
+enum { BLOCK_SIZE = 16 };
 
-static void KEY_delete(KEY *key) { delete key; }
+static KEY *KEY_new() {
+	KEY *key = EVP_CIPHER_CTX_new();
+	if (key == NULL) {
+		throw std::bad_alloc();
+	}
+	return key;
+}
+
+static void KEY_delete(KEY *key) { EVP_CIPHER_CTX_free(key); }
```

Comment:
optional: For defense-in-depth, check if key != NULL before calling CTX_free?

Reply:
[`EVP_CIPHER_CTX_free` already handles `NULL`](https://git.openssl.org/?p=openssl.git;a=blob;f=crypto/evp/evp_enc.c;h=6bb893c4d9990bc394af6098cdd387b9c9a3ceec;hb=08f876d0dea184b071a5aded4c55317e5a63c80e#l84), so I think we’re safe here.